### ISSUE TITLE
Crash when calling file upload endpoint with other content

### ISSUE THREAD

**Issue by Kulu-M** [zalando/connexion#1269]
### Description

When using a file upload endpoint but POSTing anything else than a file, for example JSON, the application [crashes on this line.](https://github.com/zalando/connexion/blob/8830d56841d29fe2ddce9d8507d60a27594cff30/connexion/decorators/validation.py#L161)

### Expected behaviour

API should not crash.

### Actual behaviour

API crashes.

### Steps to reproduce

1. Have an endpoint which accepts file upload.

```
paths:
  /upload:
    post:
      tags:
        - Upload
      operationId: mis.api.upload.upload.post
      description: This endpoint can be used to upload.
      requestBody:
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                filename:
                  type: array
                  items:
                    type: string
                    format: binary
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/uploadResult'
```

2. Call this endpoint, with a JSON body request.
3. Sadly, API crashes:
```
File "C:\...\virtualenvs\mis-KdSfneRH-py3.8\Lib\site-packages\connexion\decorators\validation.py", line 161, in wrapper data.update(dict.fromkeys(request.files, ''))  # validator expects string..
AttributeError: 'bytes' object has no attribute 'update'
```

### Additional info:

Output of the commands:

- `python --version`
Python 3.8.2

- `pip show connexion | grep "^Version\:"`
Name: connexion
Version: 2.7.0

**Comment by piyusharma95** [comment]
You need to include the suggested headers mentioned on [Swagger doc for multipart](https://swagger.io/docs/specification/describing-request-body/file-upload/)

While calling the API endpoint, include the following headers:
```
Content-Type: multipart/form-data;
Content-Disposition: form-data;
```
Let me know if this solves your issue.

**Comment by Kulu-M** [comment]
This does not resolve the issue. If you call the mentioned endpoint with a `application/json` body it results in 500 and the above mentioned error. 

This means if you have an upload endpoint, the application can be crashed by consumers - and there is nothing you can do. Except I missed something...

**Comment by joe42** [comment]
@Kulu-M I agree, this is a (minor) bug. Supplying a wrong request like this should result in an HTTP 400 BAD REQUEST error instead of a HTTP 500 error. It would be great if this could be addressed. I ran into similar additional errors with uploading files, specifically with supplying an incorrect body in combination with a multipart message, most of which we found workarounds for. Ideally bad requests should be handled gracefully by connexion since that way everyone else saves a lot of effort with individual debugging and workarounds.

**Comment by joe42** [comment]
Actually an HTTP 415 might be more appropriate here since it gives more detail on what exactly is wrong with the request.

**Comment by RobbeSneyders** [comment]
Can reproduce, although with a different error message.
